plugins/blip/basics/chartbasicsfactory.js

```diff
@@ -42,17 +43,16 @@ var BasicsChart = React.createClass({
     bgClasses: React.PropTypes.object.isRequired,
     bgUnits: React.PropTypes.string.isRequired,
     onSelectDay: React.PropTypes.func.isRequired,
+    patient: React.PropTypes.object.isRequired,
     patientData: React.PropTypes.object.isRequired,
     timePrefs: React.PropTypes.object.isRequired,
     updateBasicsData: React.PropTypes.func.isRequired,
+    updateBasicsSettings: React.PropTypes.func.isRequired,
     trackMetric: React.PropTypes.func.isRequired,
   },
   _adjustSectionsBasedOnAvailableData: function(basicsData) {
-    if (_.isEmpty(basicsData.data.reservoirChange.data)) {
-      var siteChangeSection = _.find(basicsData.sections, function(section) {
-        return section.type === 'reservoirChange';
-      });
-      siteChangeSection.active = false;
+    if (basicsData.sections.siteChanges.type !== constants.SECTION_TYPE_UNDECLARED && _.isEmpty(basicsData.data[basicsData.sections.siteChanges.type].data)) {
```

Comment:
The logic is essentially unchanged from before, with the caveat that now the section `type` is variable. I'm assuming, since you didn't explicitly state, that when you modified `patient.settings.siteChangeSource` to `reservoirChange`, you left your latest pump as Animas, Medtronic, or Tandem? That was the only way I was able to reproduce a crash.

The actual problem you encountered is a state you can't end up in through the UI. `siteChangeSource` will never* be `reservoirChange`. Reservoir changes are only displayed when the `latestPump` is OmniPod (Insulet). For OmniPods, we do not show the ability to pick what the site change source should be, so no preference is saved. For all other pumps, which have the ability to set the site change preference, the only options are cannula and tubing primes. So, a state where your pump is Animas, Medtronic, or Tandem and your `siteChangeSource` is `reservoirChange` can't currently exist. Unless you're editing the code or Mongo database by hand. 😉

_*never say never, though. I changed [this line](https://github.com/tidepool-org/tideline/pull/299/files#diff-cc5f6e0e4433fc38dfd40dcb5aab5988R218) so that we make sure to check `siteChangeSource` is not set to `reservoirChange`, if the pump is Animas, Medtronic, or Tandem. If we ever **do** give the option of reservoir changes for those pumps, we'll obviously need to refactor how the Tideline logic works overall. (e.g., filtering pump events to only show the latest pump, but that's a lot of additional processing we currently don't need)_